internal/generator/vector/output/azuremonitor/azuremonitor_test.go

```diff
@@ -0,0 +1,165 @@
+            return
+        end
+        if event.log.kubernetes.labels == nil then
+            emit(event)
+            return
+        end
+		dedot(event.log.kubernetes.namespace_labels)
+        dedot(event.log.kubernetes.labels)
+        emit(event)
+    end
+
+    function dedot(map)
+        if map == nil then
+            return
+        end
+        local new_map = {}
+        local changed_keys = {}
+        for k, v in pairs(map) do
+            local dedotted = string.gsub(k, "[./]", "_")
+            if dedotted ~= k then
+                new_map[dedotted] = v
+                changed_keys[k] = true
+            end
+        end
+        for k in pairs(changed_keys) do
+            map[k] = nil
+        end
+        for k, v in pairs(new_map) do
+            map[k] = v
+        end
+    end
+'''
+`
+		azureMonitorLogsSinkCommon = azureMonitorLogsDedot + `
+[sinks.azure_monitor_logs]
+type = "azure_monitor_logs"
+inputs = ["azure_monitor_logs_dedot"]
+customer_id = ` + customerId + `
+log_type = myType
+shared_key = ` + sharedKey
+
+		azureMonitorLogsSinkAdvance = azureMonitorLogsDedot + `
+[sinks.azure_monitor_logs]
+type = "azure_monitor_logs"
+inputs = ["azure_monitor_logs_dedot"]
+azure_resource_id = ` + azureId + `
+customer_id = ` + customerId + `
+host = ` + hostCN + `
+log_type = myType
+shared_key = ` + sharedKey
+	)
+
+	var (
+		g generator.Generator
+
+		output = loggingv1.OutputSpec{
+			Type: loggingv1.OutputTypeAzureMonitor,
+			Name: "azure_monitor_logs",
+			OutputTypeSpec: loggingv1.OutputTypeSpec{
+				AzureMonitor: &loggingv1.AzureMonitor{
+					CustomerId: customerId,
+					LogType:    "myType",
+					SharedKey:  sharedKey,
+				},
+			},
+			Secret: &loggingv1.OutputSecretSpec{
+				Name: "vector-azuremonitor-secret",
+			},
+		}
+
+		outputAdvance = loggingv1.OutputSpec{
+			Type: loggingv1.OutputTypeAzureMonitor,
+			Name: "azure_monitor_logs",
+			OutputTypeSpec: loggingv1.OutputTypeSpec{
+				AzureMonitor: &loggingv1.AzureMonitor{
+					CustomerId:      customerId,
+					LogType:         "myType",
+					SharedKey:       sharedKey,
+					AzureResourceId: azureId,
+					Host:            hostCN,
+				},
+			},
+			Secret: &loggingv1.OutputSecretSpec{
+				Name: "vector-azuremonitor-secret",
+			},
+		}
+
+		secrets = map[string]*corev1.Secret{
+			output.Secret.Name: {
+				Data: map[string][]byte{
+					"sharedKey": []byte(sharedKey),
+				},
+			},
+			outputAdvance.Secret.Name: {
+				Data: map[string][]byte{
+					"sharedKey": []byte(sharedKey),
+				},
+			},
+		}
+	)
+
+	Context("azuremonitor config", func() {
+		BeforeEach(func() {
+			g = generator.MakeGenerator()
+		})
+
+		It("should provide a valid config", func() {
+			element := Conf(output, []string{"pipelineName"}, secrets[output.Secret.Name], nil)
+			results, err := g.GenerateConf(element...)
+			Expect(err).To(BeNil())
+			Expect(results).To(EqualTrimLines(azureMonitorLogsSinkCommon))
+		})
+
+		It("should provide a valid config", func() {
```

Comment:
This text should reflect what we are testing; that is the intent of doing this kind of development.  It should be explicit so those who come after you can read the text before needing to analyze the code.